api/core/memory/token_buffer_memory.py

```diff
@@ -33,8 +34,17 @@ def get_history_prompt_messages(
 
         # fetch limited messages, and return reversed
         query = (
-            db.session.query(Message.id, Message.query, Message.answer, Message.created_at, Message.workflow_run_id)
-            .filter(Message.conversation_id == self.conversation.id, Message.answer != "")
+            db.session.query(
+                Message.id,
+                Message.query,
+                Message.answer,
+                Message.created_at,
+                Message.workflow_run_id,
+                Message.parent_message_id,
+            )
+            .filter(
+                Message.conversation_id == self.conversation.id,
```

Comment:
Hi @xuzuodong, can you explain the reason for removing `Message.answer ! = ""`?

Reply:
This is for subsequent `extract_thread_messages` method. This method accepts all messages, including the latest message, sorted from newest to oldest and begins extracting all messages that need to be returned, starting from the `parent_message_id` of the most recent message. The latest message will be popped from the returned thread message list (L60)."